Documentation/dotnet-primer.md

```diff
@@ -0,0 +1,147 @@
+
+Multiple implementations of .NET are available, based on open [.NET Standards](dotnet-standards.md) that specify the fundamentals of the platform. They are separately optimized for different app types (e.g. desktop, mobile, gaming, cloud) and support many chips (e.g. x86/x64, ARM) and operating systems (e.g. Windows, Linux, iOS, Android, OS X). Open source is also an important part of the .NET ecosystem, with multiple .NET implementations and many libraries available under OSI-approved licenses.
+
+Defining .NET Features
+======================
+
+There are a set of key features that together define .NET. Most of them are not unique on their own, but the peculiar aggregation of these features is what defines .NET as being distinct.
+
+- [Automatic memory management](garbage-collection.md)
+- Verifiable code - type and memory safety
+- Flexible native code compilation (enables JIT and AOT)
+- Fast native interop and access to native memory
+- Operating system agnostic 
+- Chip-agnostic (e.g. x86, ARM64) byte-code
+- Language-agnostic (e.g. C#, F#, VB) runtime
+- Flexible type system
+- Expressive (imperative and functional) programming languages
+- Programmable source code compiler (Roslyn)
+- Large standard class library
+- [.NET Class Libraries](class-libraries.md)
+- [Assembly format](assembly-format.md)
+- Package management
+
+TODO: Ensure that this list is correct. Write and link to a chapter for each of these defining features.
+
+.NET by Example
+===============
+
+.NET includes many features and capabilities that you will use on a daily basis for common tasks and needs. You can see these described below, with examples. Some of the same topics are discussed in more detail in documents linked above.
+
+Automatic Memory Management
+---------------------------
+
+[Garbage collection](garbage-collection.md) is the most well-known of .NET features. Developers do not need to actively manage memory, athough there are affordances to provide more information to the garbage collector (GC). C# includes the `new` keyword to allocate memory in terms of a particular type, and the `using` keyword to scope the usage of an object. The GC operates on a lazy approach to memory management, preferring application throughput to the immediate collection of memory.
+
+The following two lines both allocate. There is no analogous keyword to de-allocate the memory. The GC reclaims the memory sometime after the two variables go out of scope.
+
+	var title = ".NET Primer";
+	var list = new List<string>;
+
+Method variables normally go out of scope once a method completes, at which point they can be collected. The following example indicates to the GC that an object has gone out of scope, once the `using` statement has been completed, and the dependent `stream` object can be collected.
```

Comment:
This definitely needs a mention of `IDisposable` and that it's primarily used for wrapping unmanaged resources, otherwise it's going to confuse people trying to do `using (MyClass x = new MyClass())`